packages/extension/src/ui/features/accountTokens/tokens.state.ts

```diff
@@ -166,23 +166,24 @@ export const useFungibleTokens = (
     data: fungibleTokens,
     isValidating,
     error
-  } = useSWRImmutable(
+  } = useSWR(
```

Comment:
Question: Why do we need to change `useSWRImmutable` to `useSWR`? Do we also need to change it in the `useNonFungibleTokens`?

Reply:
Great catch, perhaps we do not need this change anymore since it will be re-computed if balance for any user tokens is updated. 
For `useNonFungibleTokens` balance is not displayed (and should be 1)

Reply:
Since the key of `useSWRImmutable` also includes the balance of the token ([link](https://github.com/alephium/extension-wallet/blob/improve-get-balance/packages/extension/src/ui/features/accountTokens/tokens.state.ts#L172)), if the key + balance is not changed, it will return the cached tokens, otherwise it will recompute.

Reply:
The key of SWR should be simple without balance, otherwise it will create too much cache due to balance updates.

I recommend we use this as the key: `[account_string, [sorted token_ids], ], 'accountNonFungibleTokens'`

Reply:
Great reasoning, will update :+1:

Reply:
Another remark: we should not refetch all of the token metadata, we should do it just for the new tokens. You could achieve this with `useRef`.

Reply:
> we should not refetch all of the token metadata,

Q: Do you mean the fetching of the token metadata [here](https://github.com/alephium/extension-wallet/blob/improve-get-balance/packages/extension/src/ui/features/accountTokens/tokens.state.ts#L189)?

Reply:
Right, it's already there!

Reply:
There is a cache for token metadata, but when we fetch a new token metadata, do we push it to the metadata cache? I don't have to time to dig

Reply:
Yeah, after we fetch the token metadata it will be stored in the `tokenStore` ([link](https://github.com/alephium/extension-wallet/blob/improve-get-balance/packages/extension/src/ui/features/accountTokens/tokens.state.ts#L191)), and will not be re-fetched next time.